### ISSUE TITLE
How does one "call" a scala class with @inject in it's constructor?

### ISSUE THREAD

**Issue by techmag** [playframework/play-mailer#56]
I've documented what I'm seeing here: http://stackoverflow.com/questions/31343401/scala-play-2-4-x-how-to-call-a-class-with-inject-annotation

**Comment by Mogztter** [comment]
When using dependency injection you rely on a framework to resolve your dependencies.
If you instantiate a class "on your own" the dependencies will not be resolved and the variable with the @Inject annotation will be undefined.

What you need to do is ask the DI framework to give you an instance (where the depencies will be resolved). See https://github.com/playframework/play-mailer/blob/master/src/main/scala/play/api/libs/mailer/MailerPlugin.scala#L28

**Comment by techmag** [comment]
Sorry - I give up - looks it's going to be easier to write everything myself than it will be to get this plugin to actually work after all. I really tried not to do that. Really tried. After all the less code I write the less code I maintain. But even after I removed all the dependencies on @Inject out of this I still can't get a simple email to go out. Sorry to have wasted your time.

**Comment by ipsq** [comment]
Because I find it incredibly important that you learn this, here is a link: http://stackoverflow.com/a/31387687/2835449

**Comment by techmag** [comment]
Yes I agree that it is important and I truly appreciate your passion on this. 

I have a time constraint issue with a project that I needed at least a proof of concept running so further development could commence however and 6 hours (without success) to get what should have been a 5 minute exercise really threw me for loop. 

I spent the balance of the day writing a mailer (in Scala) based on the Java docs and had it running in less time (Html, Text, attachments etc...). As I need way more features then this plugin provides (multi-channel with auto fail-over for starters) I likely continue down that path.  

I was actually hoping to use this as a way to start giving back to the community so I felt doubly shut out when I failed to get it to simply send a single email.

Appreciate your time and effort and the fact that you dug up my cries of anguish on StackOverflow! - first I'm going to swallow the Google documentation whole and then decide what else I need to do next.

I've build applications (most run for decades) that rely upon shared frameworks (but I "owned" both sides of the fence so to speak) so I fully understand the challenge of the problem space. I even went one step further and had shared schemas -- built my own schema migration engine to make all that pain go away too :)  

So it's rare that I hit a brick wall I don't eventually punch through -- it just seemed like this one was too much effort for too small a problem (gnat hunting with a howitzer?) at this moment in my development queue... 

At least I know what I don't know now! Always the first step...

**Comment by ipsq** [comment]
It's was confusing for me at first too. I started using Spring about 2 years ago and I did a lot of stupid things to work around some problems that hit me like unmanaged beans etc. It took me a long time (probably longer than the time I needed to learn Scala) but I'd say it was worth it. My code has gotten noticeably better.

If you have limited time on your hands, I'd stick to the primitives and not get into too much complicated stuff. It's really simple once you get the hang of it :)

They introduced this because Play was using too much static state and that is often considered bad in a web framework.
